website/pages/docs/nia/network-drivers.mdx

```diff
@@ -6,10 +6,49 @@ description: >-
   Consul-Terraform-Sync Network Drivers
 ---
 
-# Terraform
+# Network Drivers
 
-## Templating and Management
+Network drivers are used by Consul-Terraform-Sync to execute and update network infrastructure. Drivers transform Consul service level information into downstream changes by understanding and abstracting API and resource details tied to specific network infrastructure.
 
-### User Variables
+## Terraform
 
-## Automatic Terraform File Generation
+Consul-Terraform-Sync is a HashiCorp solution to Network Infrastructure Automation by bridging the networking features of Consul and infrastructure management with Terraform. The solution seamlessly embeds Terraform as a network driver to manage automation of Terraform modules. This expands the Consul ecosystem and taps into the rich features and community of Terraform and Terraform providers. Consul-Terraform-Sync automates Terraform execution of templated configuration files to carry out infrastructure changes. The auto-generated configuration leverages input variables sourced from Consul and builds on top of reusable Terraform modules published and maintained by HashiCorp partners and the community.
+
+### Understanding Terraform Automation
+
+On startup, Consul-Terraform-Sync downloads and installs [Terraform](https://www.terraform.io/downloads.html), prepares the local environment, and generates Terraform configuration files that make up the root module for each task. Terraform configuration and execution for each task is organized as separate [Terraform workspaces](https://www.terraform.io/docs/state/workspaces.html). The state files for tasks are independent of each other.
+
+~> **Note:** Although Terraform state files for task workspaces are independent, this does not guarantee the infrastructure changes from concurrent task executions are independent. Ensure that modules across all tasks are not modifying the same resource objects or have overlapping changes that may result in race conditions during automation.
+
+Within the Consul-Terraform-Sync configuration for a task, practitioners can select the desired module to run for the task as well the set of services for the module to execute on. Each task executed by the Terraform driver corresponds to an automated root module that calls the selected module in an isolated Terraform environment. Consul-Terraform-Sync will manage concurrent execution of these tasks.
+
+### Root Module
+
+The root module is simple in structure and proxies Consul service information, configuration, and other variables to the module. The content of the files that make up the root module are sourced from Consul Terraform Sync configuration, information for task's module to use as the automation playbook, and the Consul Catalog for discovering service information.
```

Comment:
we haven't really picked with/without dashes. it seems like you're mostly with dashes. maybe "Consul Terraform Sync" => "Consul-Terraform-Sync"?

Reply:
I have been manually changing these occurrences, missed this, thank you!